sdk-server-common/src/test/java/io/a2a/server/tasks/TaskManagerTest.java

```diff
@@ -175,4 +179,339 @@ public void testGetTaskNoTaskId() {
+                .build();
+        TaskArtifactUpdateEvent event1 = new TaskArtifactUpdateEvent.Builder()
+                .taskId(minimalTask.getId())
+                .contextId(minimalTask.getContextId())
+                .artifact(artifact1)
+                .build();
+        taskManager.saveTaskEvent(event1);
+
+        // Add second artifact with same artifactId (should replace the first)
+        Artifact artifact2 = new Artifact.Builder()
+                .artifactId("artifact-id")
+                .name("artifact-2")
+                .parts(Collections.singletonList(new TextPart("content 2")))
+                .build();
+        TaskArtifactUpdateEvent event2 = new TaskArtifactUpdateEvent.Builder()
+                .taskId(minimalTask.getId())
+                .contextId(minimalTask.getContextId())
+                .artifact(artifact2)
+                .build();
+        taskManager.saveTaskEvent(event2);
+
+        Task updatedTask = taskManager.getTask();
+        assertEquals(1, updatedTask.getArtifacts().size());
+        Artifact finalArtifact = updatedTask.getArtifacts().get(0);
+        assertEquals("artifact-id", finalArtifact.artifactId());
+        assertEquals("artifact-2", finalArtifact.name());
+        assertEquals("content 2", ((TextPart) finalArtifact.parts().get(0)).getText());
+    }
+
+    @Test
+    public void testMultipleArtifactsWithDifferentArtifactIds() throws A2AServerException {
+        // Test handling of multiple artifacts with different artifactIds
+        Task initialTask = minimalTask;
+        taskStore.save(initialTask);
+
+        // Add first artifact
+        Artifact artifact1 = new Artifact.Builder()
+                .artifactId("artifact-id-1")
+                .name("artifact-1")
+                .parts(Collections.singletonList(new TextPart("content 1")))
+                .build();
+        TaskArtifactUpdateEvent event1 = new TaskArtifactUpdateEvent.Builder()
+                .taskId(minimalTask.getId())
+                .contextId(minimalTask.getContextId())
+                .artifact(artifact1)
+                .build();
+        taskManager.saveTaskEvent(event1);
+
+        // Add second artifact with different artifactId (should be added)
+        Artifact artifact2 = new Artifact.Builder()
+                .artifactId("artifact-id-2")
+                .name("artifact-2")
+                .parts(Collections.singletonList(new TextPart("content 2")))
+                .build();
+        TaskArtifactUpdateEvent event2 = new TaskArtifactUpdateEvent.Builder()
+                .taskId(minimalTask.getId())
+                .contextId(minimalTask.getContextId())
+                .artifact(artifact2)
+                .build();
+        taskManager.saveTaskEvent(event2);
+
+        Task updatedTask = taskManager.getTask();
+        assertEquals(2, updatedTask.getArtifacts().size());
+        
+        // Verify both artifacts are present
+List<Artifact> artifacts = updatedTask.getArtifacts();
```

Comment:
Nitpick: Indentation. And for the very long lines below, please add some line breaks